R/proxy.R

```diff
@@ -1,28 +1,40 @@
-#' Extract underlying data
+#' Proxy and restore
 #'
 #' @description
 #'
-#' Extract the data underlying an S3 vector object, i.e. the underlying
-#' (named) atomic vector or list.
+#' `vec_proxy()` returns the memory representation of a vector. You
+#' normally don't need to implement it, except in these cases:
 #'
-#' * `vec_data()` returns unstructured data. The only attributes
-#'   preserved are names, dims, and dimnames.
+#' - S3 lists are considered scalars by default. This is the safe
```

Comment:
Wouldn't it be better to advertise using `list` in the class vector?

Reply:
Good point. I have moved this bullet to last position, and I mention how this is only an alternative to adding `"list"` to the class vector.

```
#' - S3 lists are considered scalars by default. This is the safe
#'   choice for list objects such as returned by `stats::lm()`. To
#'   declare that your S3 list class is a vector, you normally add
#'   `"list"` to the right of your class vector. Explicit inheritance
#'   from list is generally the preferred way to declare an S3 list in
#'   R, for instance it makes it possible to dispatch on
#'   `generic.list` S3 methods.
#'
#'   If you can't modify your class vector, you can implement an
#'   identity proxy (i.e. a proxy method that just returns its input)
#'   to let vctrs know this is a vector list and not a scalar.
```